core/templates/dev/head/pages/exploration-editor-page/editor-navigation/editor-navigation.directive.ts

```diff
@@ -43,15 +47,30 @@ angular.module('oppia').directive('editorNavigation', [
       controller: [
         '$rootScope', '$scope', '$timeout', '$uibModal', 'ContextService',
         'ExplorationFeaturesService', 'ExplorationRightsService',
-        'ExplorationWarningsService', 'RouterService',
+        'ExplorationWarningsService', 'ImprovementTaskService',
+        'ImprovementsDisplayService', 'RouterService',
         'SiteAnalyticsService', 'StateTutorialFirstTimeService',
         'ThreadDataService', 'UserService', 'WindowDimensionsService',
         function(
             $rootScope, $scope, $timeout, $uibModal, ContextService,
             ExplorationFeaturesService, ExplorationRightsService,
-            ExplorationWarningsService, RouterService,
+            ExplorationWarningsService, ImprovementTaskService,
+            ImprovementsDisplayService, RouterService,
             SiteAnalyticsService, StateTutorialFirstTimeService,
             ThreadDataService, UserService, WindowDimensionsService) {
+          var fetchedTasks = [];
+          ImprovementTaskService.fetchTasks().then(function(tasks) {
+            fetchedTasks = tasks;
```

Comment:
@brianrodri, are we going to do this call somewhere else as well? If yes, will it be fine to make the same call over here or we can do something better?

Reply:
Right, this should be handled by the ImprovementTaskService.